### ISSUE TITLE
bump command should not fail if commits are valid but do not trigger version bump

### ISSUE THREAD

**Issue by pwoolvett** [flakeheaven/flakeheaven#97]
Sample run: https://github.com/flakeheaven/flakeheaven/actions/runs/2396323017

**Comment by pwoolvett** [comment]
reference fix - untested

* [current](https://github.com/flakeheaven/flakeheaven/blob/69d58b83f838caec2a6e02425f98a1a297bf72f4/.github/workflows/bumpversion.yml#L44-L63)
```yaml
      - name: "Bump: commit and tag"
        run: |
          ./scripts/bump \
          && export INCOMING_TAG=`cz version --project` \
          && echo "INCOMING_TAG=${INCOMING_TAG}" >> "$GITHUB_ENV"
      - name: "Bump: Push"
        run: |
          export REMOTE_REPO="https://${{ github.actor }}:${{ secrets.FLAKEHEAVEN_PAT }}@github.com/${{ github.repository }}.git" \
          && export CURRENT_BRANCH="$(git branch --show-current)" \
          && git pull "$REMOTE_REPO" "$CURRENT_BRANCH" \
          && git push "$REMOTE_REPO" "HEAD:${CURRENT_BRANCH}" --tags
      - name: Release
        uses: softprops/action-gh-release@v1
        with:
          body_path: "incoming-changelog.md"
          tag_name: ${{ env.INCOMING_TAG }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```          

* fixed:

```yaml

      - name: "Bump: commit and tag"
        id: bump_commit
        run: |
          ./scripts/bump \
          && export INCOMING_TAG=`cz version --project` \
          && echo "INCOMING_TAG=${INCOMING_TAG}" >> "$GITHUB_ENV" \
          && echo "::set-output name=should_bump::yes" \
          || echo "::set-output name=should_bump::no"
      - name: "Bump: Push"
        if: ${{ steps.bump_commit.outputs.should_bump == 'yes'}}
        run: |
          export REMOTE_REPO="https://${{ github.actor }}:${{ secrets.FLAKEHEAVEN_PAT }}@github.com/${{ github.repository }}.git" \
          && export CURRENT_BRANCH="$(git branch --show-current)" \
          && git pull "$REMOTE_REPO" "$CURRENT_BRANCH" \
          && git push "$REMOTE_REPO" "HEAD:${CURRENT_BRANCH}" --tags
      - name: Release
        if: ${{ steps.bump_commit.outputs.should_bump == 'yes'}}
        uses: softprops/action-gh-release@v1
        with:
          body_path: "incoming-changelog.md"
          tag_name: ${{ env.INCOMING_TAG }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by pwoolvett** [comment]
fixed by [6175381](https://github.com/flakeheaven/flakeheaven/commit/6175381b29833ee6d537ca64cdaf4776fc8104f2)
